mentat/conversation.py

```diff
@@ -52,7 +55,25 @@ def get_model_response(self, config: ConfigManager) -> (str, list[CodeChange]):
         messages = self.messages.copy()
 
         code_message = self.code_file_manager.get_code_message()
-        messages.append({"role": "system", "content": code_message})
+        system_message = code_message
+
+        if self.code_map:
+            code_map_message = self.code_map.get_message()
+            if code_map_message is not None:
+                code_map_message_token_count = count_tokens(code_map_message)
+                prompt_token_count = 0
+                prompt_token_count += count_tokens(code_message)
+                for message in messages:
+                    prompt_token_count += count_tokens(message["content"])
+                token_buffer = 1000
+                token_limit = 32768 if self.allow_32k else 8192
+                token_count = (
+                    prompt_token_count + code_map_message_token_count + token_buffer
+                )
+                if token_count < token_limit:
```

Comment:
I think the user should probably know if ctags isn't added because the context isn't big enough; maybe just some yellow text?

Reply:
I agree, let's display something so the user knows what kind of map was added, and if it was added

Reply:
Good point. I'm going to add a `cprint` at the start of the session too letting the user know if the `CodeMap` is enabled at all, and if it isn't why.